src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
+                .attr("y",box_height + margin.top + padding_y)
+                .text(axis.x.label);
+        }
+
+        if (axis.y.label !== undefined) {
+            let padding_y = 15;
+            svg.append("text")
+                .attr("text-anchor","end")
+                .attr("style",label_style)
+                .attr("transform","rotate(-90)")
+                .attr("y",-margin.left + padding_y)
+                .attr("x",-margin.top - (box_height/2) + (this.getTextWidth(axis.y.label,fontStyle) / 2))
+                .text(axis.y.label);
+        }
+    }
+
+    /**
+     * Helper function to get text width to make sure text wont overlap on screen.
+     */
+    getTextWidth(text,font) {
+        const canvas  = document.createElement("canvas");
+        const context = canvas.getContext("2d");
+        context.font  = font;
+        const metrics = context.measureText("  " + text);
+        return metrics.width;
+    }
+
+    /**
+     * Creates plot's axes.
+     */
+    createAxes(axis,extremesAndDeltas,box_width,box_height,svg,focus) {
+        let {domain_x,domain_y} = this.getDomains(extremesAndDeltas,focus);
+
+        let xScale = d3.scaleLinear();
+        if (axis.x.scale !== "linear") { xScale = d3.scaleLog(); }
+
+        xScale.domain(domain_x).range([0,box_width]);
+        let xAxis = svg.append("g")
+            .attr("transform","translate(0," + box_height + ")")
+            .attr("style",label_style)
+            .call(d3.axisBottom(xScale).ticks(box_width/x_axis_label_width))
+
+        let yScale = d3.scaleLinear()
+        if (axis.y.scale !== "linear") { yScale = d3.scaleLog(); }
+
+        yScale.domain(domain_y).range([box_height,0]);
+        let yAxis = svg.append("g")
+            .attr("style",label_style)
+            .call(d3.axisLeft(yScale));
+        return {xScale:xScale,yScale:yScale,xAxis:xAxis,yAxis:yAxis};
+    }
+
+    /**
+     * Helper function calculating domains of given data.
+     */
+    getDomains(extremesAndDeltas,focus) {
+        let domain_x = [extremesAndDeltas.xMin - extremesAndDeltas.paddingX,
+            extremesAndDeltas.xMax + extremesAndDeltas.paddingX];
+        let domain_y = [extremesAndDeltas.yMin - extremesAndDeltas.paddingY,
+            extremesAndDeltas.yMax + extremesAndDeltas.paddingY];
+
+        if (focus !== undefined) {
+            if (focus.x !== undefined && focus.y !== undefined && focus.zoom !== undefined) {
+                let padding_x = extremesAndDeltas.dx * (1 / (2 * (focus.zoom)));
+                let padding_y = extremesAndDeltas.dy * (1 / (2 * (focus.zoom)));
+                domain_x = [focus.x - padding_x,focus.x + padding_x];
+                domain_y = [focus.y - padding_y,focus.y + padding_y];
+            }
+        }
+        return {domain_x,domain_y};
+    }
+
+    /**
+     * Helper function calculating extreme values and paddings to make sure data will fit nicely.
```

Comment:
Again (as with other docs) - what does it mean? Cool the data will fit nicely, but you need to describe what this function does with the data and HOW it computes the things. When someone else is reading the code, they need to understan the logic behind a function to know if it suits their needs.